api/mongowriter.go

```diff
@@ -49,3 +95,35 @@ func (mw *MongoWriter) WriteHeader(statusCode int) {
 	}
 	mw.w.WriteHeader(statusCode)
 }
+
+// ErrorBuffer returns the data stored in the error buffer.
+func (mw *MongoWriter) ErrorBuffer() []byte {
+	return mw.ew.Buffer.Bytes()
+}
+
+// ErrorStatus returns the status code with which the last call errored out,
+// if any.
+func (mw *MongoWriter) ErrorStatus() int {
+	return mw.ew.Status
+}
+
+// FailedWithWriteConflict informs us whether the MongoWriter received a MongoDB
+// WriteConflict error.
+func (mw *MongoWriter) FailedWithWriteConflict() bool {
+	return mw.ew.Status != 0 && strings.Contains(mw.ew.Buffer.String(), writeConflictErrMsg)
+}
+
+// Header implementation.
```

Comment:
maybe these should mention the interface you're implementing? or maybe also indicate why it's fine to return an empty header, these docstrings are quite dry `// WriteHeader implementation.` | `// Write implementation.`